waspc/data/Generator/templates/sdk/wasp/prisma-runtime-library.d.ts

```diff
@@ -0,0 +1,14 @@
+// Without this import, Prisma types are resolved incorrectly.
```

Comment:
@sodic this is the fix I implemented in the end, it slightly nudges the `FieldRef` type to work with our SuperJSON serialization. 

I went down the rabbit hole of recursively converting all interfaces into types using `Pick<T, keyof T>` (as suggested by some in the Github issue) but this ended up being less problematic and simple enough.

Reply:
I've updated the comment to mention the playground to make it more complete as an explanation.

> BTW, what happend with trying to fix the root of the problem, the definition of the JSONSerializable type (as attempted [here](https://hackernoon.com/mastering-type-safe-json-serialization-in-typescript)). Did Superjson make it too complex?

We concluded that `JSONSerializable` would stop users from using the Prisma field reference feature: https://www.prisma.io/docs/orm/reference/prisma-client-reference#compare-columns-in-the-same-table since the proposed approach drops fields.

Then we set on to recursively convert all interfaces to types - which in the end was a long way of doing this `d.ts` patch we have in this repo.